src/components/PostDetails.tsx

```diff
@@ -1,117 +1,144 @@
+          throw error;
+        })
+        .finally(() => setIsLoading(false));
+    }
+  }
+
+  useEffect(loadComments, [selectedPost]);
+
+  const handleDeleteComment = (commentId: number) => {
+    setIsError(false);
+    setComments(prevCommnets => prevCommnets
+      .filter(comment => comment.id !== commentId));
+
+    deleteComment(commentId)
+      .catch((error) => {
+        setIsError(true);
+        setComments(comments);
+        throw error;
+      });
+  };
+
+  const handleAddComment = ({
+    name,
+    email,
+    body,
+  }: CommentData) => {
+    setIsError(false);
+
+    createComment({
+      postId: selectedPost.id,
+      name,
+      email,
+      body,
+    })
+      .then(comment => {
+        setComments(currentComments => [...currentComments, comment]);
+      })
+      .catch((error) => {
+        setIsError(true);
+        throw error;
+      });
+  };
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            {`#${selectedPost.id}: ${selectedPost.title}`}
           </h2>
 
           <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
+            {selectedPost.body}
           </p>
         </div>
 
         <div className="block">
-          <Loader />
+          {isLoading && <Loader />}
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+          {isError && (
+            <div className="notification is-danger" data-cy="CommentsError">
+              Something went wrong
+            </div>
+          )}
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+          {!isLoading && (!comments?.length ? (
+            <p className="title is-4" data-cy="NoCommentsMessage">
+              No comments yet
+            </p>
+          ) : (
+            <>
+              <p className="title is-4">Comments:</p>
 
-          <p className="title is-4">Comments:</p>
-
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+              {comments?.map(comment => (
+                <article className="message is-small" data-cy="Comment">
```

Comment:
Better move it to common component `Comment`
Destructure `comment`
Fix all cases